src/main/java/bio/terra/stairway/Queue.java

```diff
@@ -62,47 +67,65 @@
   private final Stairway stairway;
   private final Publisher publisher;
   private final String subscriptionName;
-  private SubscriberStub subscriber;
+  private SubscriberStub subscriberStub;
 
-  public Queue(Stairway stairway, String projectId, String subscriptionId, String topicId)
+  public Queue(Stairway stairway, String projectId, String topicId, String subscriptionId)
       throws IOException {
     this.projectId = projectId;
-    this.subscriptionId = subscriptionId;
     this.topicId = topicId;
+    this.subscriptionId = subscriptionId;
+    subscriptionName = ProjectSubscriptionName.format(projectId, subscriptionId);
 
     // Create the topic and subscription
     maybeCreateTopicAndSubscription();
 
+    // Setup the publisher
+    ProjectTopicName topicName = ProjectTopicName.of(projectId, topicId);
+    publisher = Publisher.newBuilder(topicName).build();
+
+    // Build the stub for issuing pulls from the queue
     SubscriberStubSettings subscriberStubSettings =
         SubscriberStubSettings.newBuilder()
             .setTransportChannelProvider(
                 SubscriberStubSettings.defaultGrpcTransportProviderBuilder()
                     .setMaxInboundMessageSize(MAX_INBOUND_MESSAGE_BYTES)
                     .build())
             .build();
-    subscriptionName = ProjectSubscriptionName.format(projectId, subscriptionId);
-    subscriber = GrpcSubscriberStub.create(subscriberStubSettings);
-
-    ProjectTopicName topicName = ProjectTopicName.of(projectId, topicId);
-    publisher = Publisher.newBuilder(topicName).build();
+    subscriberStub = GrpcSubscriberStub.create(subscriberStubSettings);
 
     this.stairway = stairway;
   }
 
   // Create a topic and a subscription, if it doesn't exist
   private void maybeCreateTopicAndSubscription() throws IOException {
+    logger.info("Start maybeCreateTopicAndSubscription");
     ProjectSubscriptionName subscriptionName =
         ProjectSubscriptionName.of(projectId, subscriptionId);
+    logger.info("Construct topic name");
     TopicName topicName = TopicName.ofProjectTopicName(projectId, topicId);
 
-    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+    logger.info("Construct credentials");
+    GoogleCredentialsProvider credentialsProvider =
+        GoogleCredentialsProvider.newBuilder()
+            .setScopesToApply(Collections.singletonList("https://www.googleapis.com/auth/pubsub"))
+            .build();
+    logger.info("Credentials are: " + credentialsProvider);
```

Comment:
Did you mean to log credentialsProvider.getCredentials() or some other property instead of the object? And regardless, is this okay security-wise?

Reply:
Good catch. I will remove it.